flight/AIs/user_command.py

```diff
@@ -0,0 +1,73 @@
+import threading
+
+from ..drone.drone_controller import DroneController
+from .. import constants as c
+from ... import flightconfig as f
+
+def input_loop(controller):
+    """Simple command line interface to drone controller.
+
+    exit: lands and terminates program
+    land: land <priority>
+    hover: hover <duration> <priority>
+    takeoff: currently not used for safety reasons
+    move: move <FOWARD,BACK,LEFT,RIGHT> <duration> <priority>
+
+    Notes
+    -----
+    Priority is one of "high", "med", or "low".
+    """
+    while True:
+        command = raw_input('> ')
+        command = command.lower()
+        commands = command.split()
+        if len(commands) == 1 and commands[0] == "exit":
+            controller.add_exit_task()
+            return
+        elif len(commands) == 2 and commands[0] == "land":
+            priority = get_priority(commands[1])
+            controller.add_land_task(priority)
+        elif len(commands) == 3 and commands[0] == "hover":
+            priority = get_priority(commands[2])
+            controller.add_hover_task(
+                f.DEFAULT_ALTITUDE, float(commands[1]), priority)
+        elif len(commands) == 2 and commands[0] == "takeoff":
+            controller.add_takeoff_task(float(commands[1]))
+        elif len(commands) == 4 and commands[0] == "move":
+            priority = get_priority(commands[3])
+            if commands[1] == "forward":
```

Comment:
this whole if else chain should be a function

Reply:
I have added a get_direction() function.